### PR TITLE
DSODA-226 initialize google tag manager

### PR DESCRIPTION
**Pull request by bkammin**

### CODE REVIEW

File: app/instance-initializers/google-tag-manager.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import config from '../config/environment';
+
+export function initialize(/* appInstance */) {
+  if (typeof window !== 'undefined' && typeof document !== 'undefined' && config.environment !== 'test') {
+    (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+    new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+    j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+    'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
```
Comment (noslouch) [inline app/instance-initializers/google-tag-manager.js]: be sure to exclude this bit from code climate so you aren't dinged

File: app/instance-initializers/google-tag-manager.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import config from '../config/environment';
+
+export function initialize(/* appInstance */) {
+  if (typeof window !== 'undefined' && typeof document !== 'undefined' && config.environment !== 'test') {
```
Comment (noslouch) [inline app/instance-initializers/google-tag-manager.js]: you can shorten this to 
```suggestion
  if (typeof FastBoot === 'undefined' && config.environment !== 'test') {
```

File: app/routes/listen.js [hunk 3]
```diff
@@ -32,6 +34,7 @@ export default Route.extend({
       let controller = this.controllerFor('application');
       Ember.run.scheduleOnce('afterRender', this, function() {
         controller.send('updateNav');
+        get(this, 'dataLayer').sendPageView();
```
Comment (noslouch) [inline app/routes/listen.js]: shouldn't need to use a getter in the latest embers
```suggestion
        this.dataLayer.sendPageView();
```

Comment (noslouch) [reply app/routes/listen.js]: also yo might be better served with a single `routeDidChange` event handler set up in your application route, so that you aren't adding in a pageview event for every route by hand, similar to this

https://github.com/nypublicradio/gothamist-web-client/blob/6263c6212a293d931499b8cc370776015c32ab1a/app/routes/application.js#L36-L45

Comment (bkammin) [reply app/routes/listen.js]: Great, I was looking for something like this.
